packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -178,6 +178,18 @@ class RequestWrapper {
   async fetch({request}) {
     assert.atLeastOne({request});
 
+    if (this.behaviorCallbacks.requestWillFetch) {
+      for (let callback of this.behaviorCallbacks.requestWillFetch) {
+        const returnedRequest = callback({request});
+        if (returnedRequest && returnedRequest.then) {
```

Comment:
I wasn't aware of that detail.

In that case, I'd say we should just always expect the callback to return a `Promise<Request>`, since it's easy enough to for a synchronous callback to use `Promise.resolve(Request)`.